tests/ssl-opt.sh

```diff
@@ -9009,8 +9009,8 @@ run_test    "TLS 1.3: HelloRetryRequest check - gnutls" \
             "$P_CLI debug_level=4 force_version=tls13" \
             1 \
             -c "received HelloRetryRequest message" \
-            -c "HRR not supported" \
-            -c "Last error was: -0x6E00 - SSL - The handshake negotiation failed" \
+            -c "tls13 client state: MBEDTLS_SSL_CLIENT_HELLO(1)" \
+            -c "Last error was: -0x7180 - SSL - Verification of the message MAC failed" \
             -s "HELLO RETRY REQUEST was queued"
 
```

Comment:
Except `AES-256-GCM-384`, other ciphers test pass , right?
Please add other ciphers test . and add gnutls/openssl server test.

Reply:
Seems I have found the issue about AES-256-GCM-384, it's caused by: "didn't reset the sha384 script when received hrr".

Reply:
the issue about AES-256-GCM-384 should be fixed with the latest code.

Reply:
What's the root cause ? I did not catch the solution from code

Reply:
Please
- add more ciphers test . 
- add gnutls server test.

Reply:
Not addressed. 
- Add sha256 tests
- Add gnutls test

Reply:
Add SHA256 test to improve coverage

Reply:
Not addressed yet